accounts/pkg/service/v0/accounts.go

```diff
@@ -298,11 +295,12 @@ func (s Service) CreateAccount(ctx context.Context, in *proto.CreateAccountReque
 	if acc.PasswordProfile != nil {
 		if acc.PasswordProfile.Password != "" {
 			// encrypt password
-			c := crypt.New(crypt.SHA512)
-			if acc.PasswordProfile.Password, err = c.Generate([]byte(acc.PasswordProfile.Password), nil); err != nil {
+			hashed, err := bcrypt.GenerateFromPassword([]byte(acc.PasswordProfile.Password), _hashDifficulty)
+			if err != nil {
 				s.log.Error().Err(err).Str("id", id).Msg("could not hash password")
 				return merrors.InternalServerError(s.id, "could not hash password: %v", err.Error())
 			}
+			acc.PasswordProfile.Password = string(hashed)
```

Comment:
Why do we have this line? Passwords (even hashes) should never be exposed.

Reply:
It was written back before, see: https://github.com/owncloud/ocis/pull/638/files#diff-0b2098f3a82a88bc7066b4a0ccc75409L302.
I just kept the behavior.

Reply:
I think it's not exposed but gets filtered out somewhere else. Not confident about that though.

Reply:
Ooooh this is before saving it. :D Ok, nevermind then. All good like it is. Password profile gets cleared at the end of the function (same for the other commented part).